sqlmesh/core/engine_adapter/base.py

```diff
@@ -2047,7 +2047,7 @@ def temp_table(
         name = exp.to_table(name)
         # ensure that we use default catalog if none is not specified
         if isinstance(name, exp.Table) and not name.catalog and name.db:
-            name.set("catalog", value=self.default_catalog)
+            name.set("catalog", exp.to_identifier(self.default_catalog))
```

Comment:
sure

Reply:
I made this change because when running on BigQuery with our `tobiko-1` dataset, it was outputting values like `` tobiko-1.`schema`.`table` `` (i.e not quoting the `tobiko-1` which produced an error).

However, I never know when to use `to_identifier` or `parse_identifier`. I guess `parse_identifier` is more appropriate here because this value can come from the user connection config, i'll update it

Reply:
This is the main difference between the two:

```python
>>> from sqlglot import exp
>>> exp.to_identifier('"foo"')
Identifier(this="foo", quoted=True)
>>> exp.to_identifier('"foo"').sql()
'"""foo"""'
>>> exp.parse_identifier('"foo"').sql()
'"foo"'
```

Basically, `to_identifier` accepts a string and it instantiates an `Identifier` object. If it has "unsafe" chracters, then it quotes it. This happens using a regex match. On the other hand, `parse_identifier` parses the string as its name suggests.

If you came across an error after using the latter then there may be additional context around that choice, I'm not 100% sure, just observed based on hunch.

Reply:
Oh, no the error came from not doing anything at all and just shoving in the str that `self.default_catalog` returned without doing anything to turn it into an `Identifier` so it could get quoted correctly.

I think `parse_identifier` is correct here, thanks for explaining!